src/session/session_compact.c

```diff
@@ -334,6 +334,17 @@ __wt_session_compact(WT_SESSION *wt_session, const char *uri, const char *config
     session = (WT_SESSION_IMPL *)wt_session;
     SESSION_API_CALL(session, compact, config, cfg);
 
+    /* Trigger the background server thread and toggle connection statistic. */
+    if ((ret = __wt_config_gets(session, &cfg[1], "background", &cval) == 0)) {
+        /* TODO: Check background compaction server running status and trigger the mutex. */
```

Comment:
We should also check for invalid API calls. We should not have a `uri` if `background=true`.

Reply:
I've added some checks.